### PR TITLE
[TASK-73] - Product Details Page

### PR DESCRIPTION
**Pull request by thekilla0**

### CODE REVIEW

File: src/app/models/product.model.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export interface Product {
+  id: string;
+  name: string;
+  description: string;
+  price: number;
+  // Add any other fields you need
+}
```
Comment (Theinfinix500) [inline src/app/models/product.model.ts]: you can check the product model from this link https://github.com/Theinfinix500/ecom-lab-backend/blob/develop/src/products/dto/create-product.dto.ts

File: src/app/product-detail/product-detail.component.scss [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+// .product-detail h2 {
+//   font-size: 24px;
+//   margin-bottom: 10px;
+// }
+
+// .product-detail p {
+//   font-size: 18px;
+//   margin: 5px 0;
+// }
```
Comment (Theinfinix500) [inline src/app/product-detail/product-detail.component.scss]: remove comments

File: src/app/product-detail/product-detail.component.scss [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+.wrapper {
+  display: flex;
+  flex-direction: column;
+}
```
Comment (Theinfinix500) [inline src/app/product-detail/product-detail.component.scss]: add gap: 1rem;

File: src/app/product-detail/product-detail.component.scss [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+.products {
+  padding-bottom: 10px;
+}
```
Comment (Theinfinix500) [inline src/app/product-detail/product-detail.component.scss]: remove padding

File: src/app/product-detail/product-detail.component.scss [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+.remove-s {
+  padding-left: 110px;
```
Comment (Theinfinix500) [inline src/app/product-detail/product-detail.component.scss]: remove padding

File: src/app/product-detail/product-detail.component.scss [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  display: flex;
+  flex-flow: row nowrap;
+  flex: 1;
+  justify-content: space-evenly;
```
Comment (Theinfinix500) [inline src/app/product-detail/product-detail.component.scss]: remove these

File: src/app/product-detail/product-detail.component.scss [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+  flex-direction: column;
+}
+
+.products {
+  padding-bottom: 10px;
+}
+
+.remove-s {
+  padding-left: 110px;
+  display: flex;
```
Comment (Theinfinix500) [inline src/app/product-detail/product-detail.component.scss:8]: add gap: .5rem;
    align-items: center;

File: src/app/product-detail/product-detail.component.scss [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+  flex-flow: row nowrap;
+  flex: 1;
+  justify-content: space-evenly;
+}
+
+.small-image {
+  padding-right: 20px;
+}
```
Comment (Theinfinix500) [inline src/app/product-detail/product-detail.component.scss]: replace this rule with:

::ng-deep p-image img {
    max-width: 50px;
    height: 100%;
    object-fit: cover;
    aspect-ratio: 1;
}

File: src/app/product-detail/product-detail.component.scss [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+
+b {
+  padding-left: 6px;
+}
```
Comment (Theinfinix500) [inline src/app/product-detail/product-detail.component.scss]: replace b with p and give it these styles:
line-height: 30px;
margin: 0;

File: src/app/product-detail/product-detail.component.scss [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+
+.products-details {
+  display: flex;
+  flex-direction: row;
+}
+
+.big-title {
+  display: flex;
+  flex-direction: column;
+  padding-bottom: 24px;
```
Comment (Theinfinix500) [inline src/app/product-detail/product-detail.component.scss]: remove padding

File: src/app/product-detail/product-detail.component.scss [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+.products-details {
+  display: flex;
+  flex-direction: row;
+}
+
+.big-title {
+  display: flex;
+  flex-direction: column;
```
Comment (Theinfinix500) [inline src/app/product-detail/product-detail.component.scss:31]: add property:
flex: 1;

File: src/app/product-detail/product-detail.component.scss [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+.go-back {
+  border: none;
+  background: none;
+}
+
+.products-details {
+  display: flex;
+  flex-direction: row;
```
Comment (Theinfinix500) [inline src/app/product-detail/product-detail.component.scss:25]: add property:
gap: 1rem;

File: src/app/product-detail/product-detail.component.html [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+        width="70"
+        height="50" />
+    </div>
+    <div class="big-title">
+      <p><b>A Walk Amongst Friends - Canvas Print</b></p>
+      <span>Product Details</span>
+    </div>
+    <div class="remove-s">
+      <p-button label="Submit" icon="pi pi-trash" />
+      <p-button label="Save" [disabled]="true" />
```
Comment (Theinfinix500) [inline src/app/product-detail/product-detail.component.html]: make those rounded with [rounded]="true"

File: src/app/product-detail/product-detail.component.html [hunk 14]
```diff
@@ -0,0 +1,76 @@
+<div class="wrapper">
+  <div class="products">
+    <button class="go-back" [text]="true" pButton>
+      <i class="pi pi-arrow-left"><b>Products</b></i>
+    </button>
```
Comment (Theinfinix500) [inline src/app/product-detail/product-detail.component.html]: remove <i tag and use the icon="pi pi-check" as button input.
check [docs](https://primeng.org/button#icons)

**Review by Theinfinix500 — CHANGES_REQUESTED** [review]

File: src/app/product-detail/product-detail.component.html (hunk 14 above)
Comment (Theinfinix500) [reply src/app/product-detail/product-detail.component.html:5]: remove i tag*

File: src/app/product-detail/product-detail.component.scss (hunk 8 above)
Comment (thekilla0) [reply src/app/product-detail/product-detail.component.scss]: i did face an error when i commited in the (::ng-deep p-image img), it says that it is an unkown pseudo element.

**Review by Theinfinix500 — APPROVED** [review]

**Pull request merged**